src/Sql/Expression.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+declare(strict_types=1);
+
+namespace SimPod\ClickHouseClient\Sql;
+
+use function Safe\sprintf;
+
+final class Expression
+{
+    /** @var string */
+    public $innerExpression;
+
+    private function __construct(string $expression)
+    {
+        $this->innerExpression = $expression;
+    }
+
+    public static function new(string $expression) : self
+    {
+        return new self($expression);
+    }
+
+    /**
+     * @param mixed $value
+     */
+    public static function fromTemplateAndValue(string $template, $value) : self
```

Comment:
přijde mi škoda nepoužít tu variadic funkce ... byl by pak jeden konstruktor a bylo by to univerzálnější

Reply:
to by vypadalo ted asi hezky, ale komplikuju si tim BC do budoucna. radsi vic konstruktoru, takze vyssi flexibilnost pro zmeny v api

Reply:
nový konstruktor pro nový use case můžeš udělat vždy ... ale mít jeden pro dva běžný usecases, když to jde, mi dává větší smysl no

Reply:
nový konstruktor pro nový use case můžeš udělat vždy ... ale mít jeden pro dva běžný usecases, když to jde, mi dává větší smysl no

Reply:
to je ten problem: jeden pro dva use cases

dva usecases = dve oddeleni

aby se navzajem nemohly ovlivnit/zamenit etc. zas solid

Reply:
jak chceš, nemám s tím vyloženě probleḿ, ale přijde mi fakt na hovno na něco co se používá tak často mít takhle dlouhý jméno ... a ještě to má omezenou funkcionalitu oproti tomu co jsem navrhoval já

Reply:
metoda by mela mit jen jeden jasne danej zpusob pouziti no

Reply:
variadic jsem musel pridat